test/Interop/Cxx/constructor/copy-constructor-irgen.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: %target-swift-frontend -enable-cxx-interop -I %S/Inputs %s -emit-ir | %FileCheck %s
+
+import Basic
+
+// CHECK-LABEL: define swiftcc void @"$s4main4testSo30HasUserProvidedCopyConstructorVyF"
+// CHECK: [[OBJ:%.*]] = alloca %TSo30HasUserProvidedCopyConstructorV
+// Make sure we bitcast it from the Swift type to the C++ type.
+// CHECK: [[CXX_ARG:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* %0 to %struct.HasUserProvidedCopyConstructor*
+// CHECK: [[CXX_OBJ:%.*]] = bitcast %TSo30HasUserProvidedCopyConstructorV* [[OBJ]] to %struct.HasUserProvidedCopyConstructor*
+// Make sure we call the destructor.
+// CHECK: call void @_ZN30HasUserProvidedCopyConstructorC1ERKS_(%struct.HasUserProvidedCopyConstructor* [[CXX_ARG]], %struct.HasUserProvidedCopyConstructor* [[CXX_OBJ]])
+// CHECK: ret void
+public func test() -> HasUserProvidedCopyConstructor {
+  let obj = HasUserProvidedCopyConstructor(42)
+  return obj
+}
+
+// CHECK-LABEL: define linkonce_odr void @_ZN30HasUserProvidedCopyConstructorC1ERKS_
+// CHECK: call void @_ZN30HasUserProvidedCopyConstructorC2ERKS_
```

Comment:
Do we really need to be checking Clang's codegen here?

Besides, this will break on Microsoft ABIs.

Reply:
Fair enough. Done.